docs/1.0. Authorization Practice.md

```diff
@@ -31,10 +33,68 @@ the Resource server rejecting the request with a '403' HTTP response code and a
 error value of `insufficient_scope` as specified in [RFC 6750][RFC-6750]. All IS-04 resources can be related back to a
 single Node ID by following the [IS-04 Referential Integrity][] guide.
 
+#### WebSocket Authorization
+
+When creating an [IS-04 Registry][] WebSocket subscription, whether by using a POST to the `/subscriptions` resource
+or by using an `Upgrade` HTTP header, the JSON Web Token used MUST be validated to ensure it contains a write claim
+matching the subscriptions path, for example:
+
+```
+"x-nmos-query": {
+  "write": ["subscriptions"],
+  "read": ["*"]
+}
+```
+
+When connecting to an [IS-04 Registry][] WebSocket subscription, the JSON Web Token used MUST be validated to ensure it
+contains a read claim matching all data which may be returned according to the subscription's `resource_path`
+configuration. Note that the `resource_path` includes a leading slash where token claims do not. The `ws_href` path
+MUST NOT be validated against the token claims.
+
+For example, access to the following subscription would require token claims as follows:
+
+```
+{
+  "max_update_rate_ms": 100,
+  "resource_path": "/receivers",
+  "params": {},
+  "persist": false,
+  "secure": false,
+  "ws_href": "wss://172.29.80.52:8870/ws/?uid=6a52dbd5-a737-4c4e-823f-909ade8f8bf4",
+  "id": "6a52dbd5-a737-4c4e-823f-909ade8f8bf4"
+}
+```
+
+```
+"x-nmos-query": {
+  "read": ["receivers/*"]
+}
+```
+
+### IS-07 - Event and Tally
+
+#### WebSocket Authorization
+
+When connecting to an [IS-07][] WebSocket, the JSON Web Token used MUST be validated to ensure it contains a read claim
+matching all Source data which may be returned by the WebSocket. These Source paths match those used by the IS-07 API.
+
+For example, access to the IS-07 Source IDs '9f463872-9621-4939-aa3a-dc3c82d8578b' and
+'7f87027c-ebb4-4640-b878-14952915249a' only would require token claims as follows.
+
+```
+"x-nmos-events": {
+  "read": ["sources/9f463872-9621-4939-aa3a-dc3c82d8578b",
+           "sources/7f87027c-ebb4-4640-b878-14952915249a"]
+}
+```
+
```

Comment:
Worth adding this? (Minus the `###` - I couldn't work out how to make a suggestion with an embedded code block!)
```suggestion

Alternatively, access to all IS-07 Source IDs may be requested with a token claim as follows.

###```
"x-nmos-events": {
  "read": ["sources/*"]
}
###```

```